easybuild/tools/module_generator.py

```diff
@@ -1110,6 +1123,11 @@ def get_description(self, conflict=True):
         for line in self._generate_whatis_lines():
             whatis_lines.append("whatis(%s%s%s)" % (self.START_STR, self.check_str(line), self.END_STR))
 
+        provide_list = self._generate_provides_list()
+        if self.modules_tool.supports_extensions and provide_list:
+            provide_list_str = ', '.join(['"%s"' % x for x in provide_list])
+            lines.extend(['', 'extensions(%s)' % provide_list_str])
```

Comment:
We have two options here:

* either we include this if the Lmod version used at installation time is recent enough (which is what you're doing here)
* or we add a conditional in the module file itself, using `LmodVersion()` (that's what Robert recommended)

The benefit of the latter approach is that the module files will still work when loaded with an older Lmod version, and there's basically no overhead in using the conditional.

Reply:
It's a design choice, I don't have a strong preference for either.

But adding support in a module file for something that doesn't work with the current Lmod seems a bit weird to me? Especially as it's pretty easy to regenerate a module.

Reply:
Also, it might surprise people. You update Lmod and suddenly the output of av etc changes

Reply:
It's a protection mechanism too though. You may be installing software on a system with a slightly more recent version of Lmod (8.2+), and then consume the modules elsewhere with a slightly older Lmod (<8.2).
That wouldn't be very nice either... It's certainly less serious than being a bit careful and only using `extensions` conditionally.

Output of `ml av` will change anyway as soon as you install something with extensions after updating to Lmod 8.2, and you're probably not going to notice straight away (I certainly rarely check the generated module files mannually ;) ).